src/wp-includes/script-loader.php

```diff
@@ -2232,6 +2232,52 @@ function wp_common_block_scripts_and_styles() {
 	do_action( 'enqueue_block_assets' );
 }
 
+/**
+ * Enqueues the global styles defined via theme.json.
+ *
+ * @since 5.8.0
+ *
+ * @return void
+ */
+function wp_enqueue_global_styles() {
+	if ( ! WP_Theme_JSON_Resolver::theme_has_support() ) {
```

Comment:
In cases where there is link color enabled add_theme_support('experimental-link-color'); we need to output the preset classes, should we have a condition for that here?

Reply:
I'm not familiar with prior context on how to deal with this so I'd welcome advice. My thinking is that we should do the following:

- We shouldn't port `experimental-link-color` to WordPress core. Themes that want that should use enable it via `theme.json`.
- In the plugin, I'd expect we remove support for `experimental-link-color` at some point. Probably when the minimum WordPress version for the plugin is 5.8.

Thoughts?

Reply:
Yes this seems like a plan :